### ISSUE TITLE
Indeterminate / mixed / blank / empty value flags for scalar widget elements

### ISSUE THREAD

**Issue by nekitu** [ocornut/imgui#6865]
This issue was brought up before by other user(s).
InputScalar functions do not allow empty value display, this is critical for property inspector UIs when multiple objects/records are selected in the application. I dont want to copy and modify the entire InputScalar code/logic, this should be a feature in those functions, as flags for each element ImGuiInputTextFlags_EmptyX, ImGuiInputTextFlags_EmptyY...etc.
For example UnrealEditor shows "Multiple Values" and Unity shows "---" when there is a blank/indeterminate scalar.
Would be greatly welcomed if this feature is added to the numeric widgets, thanks!

**Comment by ocornut** [comment]
Linking to #5518, we'd use something like the `ImGuiItemFlags_MixedValue` flag for this.

**Comment by nekitu** [comment]
Thank you for the fast response, I will look into it!

**Comment by nekitu** [comment]
Yeah, checked the code and its just for the checkbox. In any case, this is much needed for the scenario above, ideally per scalar element, since some might be the same, some can be different, I will attach Unreal and Unity how they show it.
![image](https://github.com/ocornut/imgui/assets/10060056/7fc6c357-780a-49e1-9e2b-9ed70abbcaee)
![image](https://github.com/ocornut/imgui/assets/10060056/6f05a0af-c016-453a-82cb-e4de9e03528f)

**Comment by GamingMinds-DanielC** [comment]
I have recently implemented custom widgets for this exact purpose, also based on Unreal and Unity. Some artists didn't like that the values couldn't be seen anymore, so I had to display the value of the last selected item grayed out (to indicate multiple values) instead. I would share the code, but it has a few dependencies into custom code that stand in the way. Especially rotation widgets (they only get quaternions, but can optionally convert them to Euler angles for display and editing) and scaling (can switch between uniform and nonuniform modes based on flags of selected objects).

![Widgets](https://github.com/ocornut/imgui/assets/77451134/36f40759-5352-47c9-97cf-c197fc1a1655)

Together with some other extensions, this gets very specific very fast, maybe too specific to integrate this into ImGui directly. In my opinion a small library with ImGui as its only dependency would make more sense, there might even be a few out there already. And doing it in a separate library would point out interface bottlenecks, ultimately leading to more flexible interfaces and therefore more options for custom widgets.

**Comment by nekitu** [comment]
yeah, I would not agree with the artists on the last value shown in gray :), should be specific that its indeterminate hence empty/hyphens/explicit text. We just need that Mixed logic Omar started on the checkbox to be implemented for the other widgets, the rest of the logic will be app dependent. Just need those flags so those widgets can show indeterminate state.

**Comment by ocornut** [comment]
I have pushed a solution for this, see https://github.com/ocornut/imgui/issues/5518#issuecomment-5573724578

Please comment there or open a new issue if you feel further improvements are needed (namely, @GamingMinds-DanielC suggestion is not yet supported).
